### ISSUE TITLE
[java] A false negative about the rule StringBufferInstantiationWithChar

### ISSUE THREAD

**Issue by Zustin** [pmd/pmd#4042]
**Affects PMD Version:**
6.47.0+
7.0.0

**Rule:StringBufferInstantiationWithChar**

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/latest/pmd_rules_java_errorprone.html#stringbufferinstantiationwithchar>

**Description:**
Hi, I found a false negative about the rule StringBufferInstantiationWithChar, please read the following minimized code example. In line 3, `builder` is initialized by a character, but `ch` could converted into an integer. Hence, I think this is a false negative, and maybe we should analyze variable type about this rule.

**Code Sample demonstrating the issue:**

```java
public void foo() {
    char ch = 'c';
    StringBuilder builder = new StringBuilder(ch);  // should report a warning in this line
}
```

**Expected outcome:A warning**

PMD should report a violation at line 3, but doesn't. This is a false-negative.

**Running PMD through:** *[Maven]*
